docs/04-BasicUsage.md

```diff
@@ -174,37 +174,24 @@ Next thing to render is a text input for `email` field:
 <input
   value={form.input.email}
   disabled={form.submitting}
-  onBlur={_ => form.blurEmail()}
-  onChange={event => {
-    let value = event->ReactEvent.Form.target##value;
-    form.updateEmail(input => {...input, email: value});
-  }}
+  onBlur={form.blurEmail}
+  onChange={
+    form.updateEmail((~target, input) => {
+      ...input,
+      email: target##value,
+    })
+  }
 />
 ```
 
 The value of the field is exposed via `form.input` record. For extra safety, we disable all inputs during form submission using `form.submitting` property which is of boolean type. The next 2 functions are very important:
-1. `form.blurEmail: unit => unit`: must be triggered from `onBlur` handler of an input field
-2. `form.updateEmail: (input => input) => unit`: must be triggered from `onChange` handler of an input field. It takes a function as an argument which takes the current form `input` and must return updated `input` record.
+1. `form.blurEmail: ReactEvent.Focus.t => unit`: must be triggered from `onBlur` handler of an input field
```

Comment:
This type signature is much better! The intended use is encoded in the function type sig instead of a `unit => unit`